packages/react/slot/src/slot.tsx

```diff
@@ -10,43 +10,26 @@ interface SlotProps extends React.HTMLAttributes<HTMLElement> {
 }
 
 /* @__NO_SIDE_EFFECTS__ */ export function createSlot(ownerName: string) {
-  const SlotClone = createSlotClone(ownerName);
   const Slot = React.forwardRef<HTMLElement, SlotProps>((props, forwardedRef) => {
     const { children, ...slotProps } = props;
-    const childrenArray = React.Children.toArray(children);
-    const slottable = childrenArray.find(isSlottable);
-
-    if (slottable) {
-      // the new element to render is the one passed as a child of `Slottable`
-      const newElement = slottable.props.children;
-
-      const newChildren = childrenArray.map((child) => {
-        if (child === slottable) {
-          // because the new element will be the one rendered, we are only interested
-          // in grabbing its children (`newElement.props.children`)
-          if (React.Children.count(newElement) > 1) return React.Children.only(null);
-          return React.isValidElement(newElement)
-            ? (newElement.props as { children: React.ReactNode }).children
-            : null;
-        } else {
-          return child;
-        }
-      });
-
-      return (
-        <SlotClone {...slotProps} ref={forwardedRef}>
-          {React.isValidElement(newElement)
-            ? React.cloneElement(newElement, undefined, newChildren)
-            : null}
-        </SlotClone>
-      );
+    const slottableElement = findSlottableElement(children);
+    const slottableElementRef = slottableElement ? getElementRef(slottableElement) : undefined;
+    // fine in RSC as it's a React.useCallback under the hood
+    const composedRefs = useComposedRefs(forwardedRef, slottableElementRef);
```

Comment:
This broke in the previous attempts because Slot was changed to a `'use client'` boundary. We can compose them without that, and all is well :)